src/EventListener/InitializeSystemListener.php

```diff
@@ -285,14 +290,28 @@ private function validateInstallation(Config $config, Request $request = null)
         }
 
         // Show the "insecure document root" message
-        // FIXME: add unit tests for this as soon as die_nicely is an exception
         if (!in_array($request->getClientIp(), ['127.0.0.1', 'fe80::1', '::1']) && '/web' === substr($request->getBasePath(), -4)) {
-            die_nicely('be_insecure', 'Your installation is not secure. Please set the document root to the <code>/web</code> subfolder.');
+            throw new InsecureInstallationHttpException();
         }
 
         // Show the "incomplete installation" message
         if (!$config->isComplete()) {
-            die_nicely('be_incomplete', 'The installation has not been completed. Open the Contao install tool to continue.');
+            // FIXME: This maybe should be removed when localconfig.php is not mandatory anymore.
+            throw new IncompleteInstallationHttpException();
+        }
+    }
+
+    /**
+     * Ensure we do not have any error level active that could be harmful to the Contao 3 Framework.
+     */
+    private function sanitizeErrorHandling()
+    {
+        // Check if we have "simple errors" and make sure to kill them as the legacy code will otherwise break.
+        // This happens (at least) on console.
+        $level    = error_reporting();
+        $newLevel = $level & ~(E_NOTICE | E_STRICT | E_DEPRECATED);
+        if ($level & $newLevel) {
+            error_reporting($newLevel);
```

Comment:
It is the result of heavy testing on my machine which is also the reason of killing the configurable error levels.
If we do not kill at least these, the core did not boot - will get even worse with extensions involved which cause recoverable errors due to improper type hints etc.
The whole installation will die then.
We could of course change our code to be safe against all these errors by putting heavy effort into fixing them but it is not worth the hassle.
I only dislike that we will loose the benefit of "enforcing" the users to write proper code for new extensions (code not emitting notices etc.).
My idea was to write an error handler for dev mode which will check where the notice originates from and accordingly either ignore it or not. Should not be a big problem as new code should never end up in there and therefore not slow down the system.
If you agree, I will provide an additional PR for this then.